kernel/src/create_table.rs

```diff
@@ -0,0 +1,220 @@
+use std::collections::HashMap;
+
+use chrono::Utc;
+use url::Url;
+
+use crate::actions::{
+    get_log_commit_info_schema, get_log_metadata_schema, get_log_protocol_schema, CommitInfo,
+    Metadata, Protocol,
+};
+use crate::path::ParsedLogPath;
+use crate::schema::StructType;
+use crate::table_features::{
+    ReaderFeature, WriterFeature, SUPPORTED_READER_FEATURES, SUPPORTED_WRITER_FEATURES,
+};
+use crate::{DeltaResult, Engine, IntoEngineData};
+
+#[cfg(test)]
+mod tests;
+
+// default to 3, 7 with all our supported table features
+const DEFAULT_MIN_READER_VERSION: i32 = 3;
+const DEFAULT_MIN_WRITER_VERSION: i32 = 7;
+
+const DEFAULT_CREATE_TABLE_OPERATION: &str = "CREATE";
```

Comment:
Is there a case when this isn't `CREATE`?

Reply:
well we don't actually _require_ it to be anything. I think by convention it _should_ be CREATE but the engine is free to put whatever they want there.